File: CosmereScadrial/.scripts/Metals/MetalInfo.ts
export class MetalInfo {
  public Allomancy: MetalAllomancyInfo
  public Color: [number, number, number];
  public DefName?: string;
  public Feruchemy: MetalFeruchemyInfo;
  public GodMetal: boolean;
  public MaxAmount: number = 100;
  public Name: string;
  constructor(self: Partial<MetalInfo>) {
    Object.assign(this, self);
  }
}

export class MetalAllomancyInfo {
  public Axis: string;
  public Description: string;
  public Group: string;
  public Polarity: string;
  public UserName: string;
  constructor(self: Partial<MetalAllomancyInfo>) {
    Object.assign(this, self);
  }
}

export class MetalFeruchemyInfo {
  public Description: string;
  public Group: string;
  public UserName: string;
  constructor(self: Partial<MetalFeruchemyInfo>) {
    Object.assign(this, self);
  }
}

File: CosmereScadrial/CosmereScadrial/MetalInfo.cs
#nullable disable
using UnityEngine;

namespace CosmereScadrial;

public class MetalRegistryData {
    public MetalInfo[] metals;
}

public class MetalInfo {
    public MetalAllomancyInfo allomancy;
    public Color color;
    public string defName;
    public MetalFeruchemyInfo feruchemy;
    public bool godMetal;
    public float maxAmount = 100f;
    public string name;
}

public class MetalAllomancyInfo {
    public string axis;
    public string description;
    public string group;
    public string polarity;
    public string userName;
}

public class MetalFeruchemyInfo {
    public string description;
    public string group;
    public string userName;
}